common/src/main/java/org/broadleafcommerce/common/resource/service/ResourceMinificationService.java

```diff
@@ -20,24 +20,43 @@
 import org.springframework.core.io.Resource;
 
 /**
- * Service responsible for interacting with YUI-compressor to minify JS/CSS resources.
+ * Service responsible for interacting with YUI-compressor or Google Closure Compiler to minify JS/CSS resources.
+ *
+ * YUI-compressor is controlled via the following properties:
+ *
+ * <ul>
+ *  <li>minify.enabled - whether or not to actually perform minification</li>
+ *  <li>minify.linebreak - if set to a value other than -1, will enforce a linebreak at that value</li>
+ *  <li>minify.munge - if true, will replace variable names with shorter versions</li>
+ *  <li>minify.verbose - if true, will display extra logging information to the console</li>
+ *  <li>minify.preserveAllSemiColons - if true, will never remove semi-colons, even if two in a row exist</li>
+ *  <li>minify.disableOptimizations - if true, will disable some micro-optimizations that are performed</li>
+ * </ul>
+ *
+ * The Google Closure Compiler uses to minify only JS resources.
+ * To use it instead of the YUI-compressor, add the property: minify.closure.compiler.enabled=true
+ *
+ * Compiler is controlled via the following properties:
+ * <ul>
+ *  <li>minify.closure.compiler.languageIn - Sets ECMAScript version to use for the input.
+ *  Options: ECMASCRIPT3, ECMASCRIPT5, ECMASCRIPT5_STRICT, "
+ * "ECMASCRIPT6_TYPED (experimental), ECMASCRIPT_2015, ECMASCRIPT_2016, "
+ * "ECMASCRIPT_2017, ECMASCRIPT_NEXT</li>
+ *
+ *  <li>minify.closure.compiler.languageOut - Sets ECMAScript version to use for the output.
+ *  Options: ECMASCRIPT3, ECMASCRIPT5, ECMASCRIPT5_STRICT, "
+ * "ECMASCRIPT6_TYPED (experimental), ECMASCRIPT_2015, ECMASCRIPT_2016, "
+ * "ECMASCRIPT_2017, ECMASCRIPT_NEXT, NO_TRANSPILE</li>
+ *
+ *  <li>minify.closure.compiler.warningLevel - Warnings level. Possible values: QUIET, DEFAULT, VERBOSE</li>
+ * </ul>
```

Comment:
Since we're creating multiple implementations of this class it's probably better to put the docs that are relevant to that implementation there instead of in the interface. The interface should have minification library agnostic information about what this interface is used for and what the methods do.